src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -947,7 +947,7 @@ private static ulong CurlReceiveHeadersCallback(IntPtr buffer, ulong size, ulong
                                 {
                                     response.Content.Headers.TryAddWithoutValidation(headerName, headerValue);
                                 }
-                                else if (easy._isRedirect && string.Equals(headerName, HttpKnownHeaderNames.Location, StringComparison.OrdinalIgnoreCase))
+                                else if (string.Equals(headerName, HttpKnownHeaderNames.Location, StringComparison.OrdinalIgnoreCase))
```

Comment:
If a Location header is set on a HTTP-200 response, what happens?  curl calls it an error? we redirect?

What about the Windows version?

(It'd make an interesting test case :smile:)